aspnetcore/performance/response-compression.md

```diff
@@ -0,0 +1,170 @@
+
+The headers involved in requesting, sending, caching, and receiving compressed content are described below.
+
+Header | Role
+--- | ---
+`Accept-Encoding` | Sent by the client to the server to indicate which types of content encoding are acceptable.
+`Content-Encoding` | Sent from the server to the client to indicate the encoding of the content in the payload.
+`Content-Length` | When compression occurs, the the `Content-Length` header is removed, since the body content changes when the response is compressed.
+`Content-MD5` | When compression occurs, the `Content-MD5` header is removed, since the body content has changed and the hash is no longer valid.
+`Content-Type` | Specifies the MIME type of the content. MIME types are specified on the server for encodable content, so this information is used by the server to determine if a response can be compressed. The middleware includes a set of [default MIME types](#mime-types) that it will encode, but you can replace or add MIME types.
+`Vary` | When sent by the server with a value of `Accept-Encoding` to clients and proxies, it indicates that they should cache (vary) responses based on the value of the `Accept-Encoding` header of the request. The result of returning content with the `Vary: Accept-Encoding` header is that both compressed and uncompressed responses will be cached separately.
+
+## When to use Response Compression Middleware
+Use Response Compression Middleware when you're unable to use the [Dynamic Compression module](https://www.iis.net/overview/reliability/dynamiccachingandcompression) in IIS on Windows Server, the [Apache mod_deflate module](http://httpd.apache.org/docs/current/mod/mod_deflate.html), [NGINX Compression and Decompression](https://www.nginx.com/resources/admin-guide/compression-and-decompression/), or your application is hosted on [WebListener server](xref:fundamentals/servers/weblistener). The main reason to use the server-based response compression technologies in IIS, Apache, or Nginx is that the performance of the middleware probably won't match that of the server modules. 
+
+## Package
+To include the middleware in your project, add a reference to the `Microsoft.AspNetCore.ResponseCompression` package. The middleware is available for projects that target .NETFramework 4.5.1 or .NETStandard 1.3 or higher.
+
+## Configuration
+You can explore the features of the Response Compression Middleware with the [response compression sample application](https://github.com/aspnet/Docs/tree/master/aspnetcore/performance/response-compression/sample). The sample illustrates the compression of application responses using GZip and custom compression providers. It also shows you how to add a MIME type to the default list of MIME types for compression.
+
+### Using defaults
```

Comment:
> Since the previous section will be gone

I misunderstood your remark in the last round. You want to change the outline so that the next level sections under *Configuration* move up a level. The new outline will be ...

* Response Compression (intro)
* When to use it
* Package
* Default Configuration
* Providers
  * GZipCompressionProvider
  * Custom Providers
* MIME types
* Compression with TLS
* Adding Vary
* Dropping IIS compression
* Troubleshooting

... however, I want to take those two sentences about the sample and move them to line 21 (the introductory paragraph) to tell the reader a little more about the sample and what it does.

Also, see my comment below before answering on this, because there's another outline problem regarding the location of the sample text within the doc.